### PR TITLE
Add gradient fns for log and abs

### PR DESCRIPTION
**Pull request by joelshepherd**
This adds gradient functions for `log` and `abs`.

I have used and tested these locally for cross entropy and mean absolute error loss functions. If you would like having these in your library too, I am happy to upstream [them](https://github.com/joelshepherd/ml/blob/main/lib.ts#L87) from my experiment repo too.

### CODE REVIEW

**Comment by bwasti** [comment]
thanks! Abs looks great but I think the gradient for log is failing:
https://app.circleci.com/pipelines/github/facebookresearch/shumai/667/workflows/d449cbda-f5e8-463a-83cf-35c1f9588c8f/jobs/1844/parallel-runs/0/steps/0-113


> cross entropy and mean absolute error loss functions.

oh yea, we definitely need better loss function coverage! would really appreciate that :)

**Comment by joelshepherd** [comment]
I can address the failing logs by moving the allowed error up to above `0.1`. The amount of error increases with the absolute value of the gradient. Perhaps we allow a error % for this function?

Though doing so still doesn't seem to prevent the finite diff from producing `nan`s (at least in the console). I haven't quite worked out what is causing them.

image

FYI: I think you may have "expected" and "actual" around the wrong way here. Should the "finite diff" be the expected value?

**Comment by joelshepherd** [comment]
I have fixed the tests by moving the input tensor away from log's undefined 0. Similar to what I do for the loss function.
